lib/jsonapi/include_directive.rb

```diff
@@ -12,6 +12,8 @@ module JSONAPI
   # @example 'posts.**' # => Include related posts, and all the included
   #   posts' related resources, and their related resources, recursively.
   class IncludeDirective
+    attr_reader :options, :hash
```

Comment:
Instead of exposing the hash, you could just make `IncludeDirective` enumerable by
```ruby
def each
  @hash.each
end
```
then
```ruby
def merge!(other)
  other.each do |key, val|
    @hash[key] ||= IncludeDirective.new({}, options)
    @hash[key].merge!(val)
  end
end

def merge(other)
  id = IncludeDirective.new(to_hash, options)
  id.merge!(other)

  id
end
```